src/components/data/DataNavigation.js

```diff
@@ -0,0 +1,460 @@
+ *
+ */
+
+import React, { useEffect, useState } from "react";
+import { build, formatMessage, withI18N } from "@cyverse-de/ui-lib";
+import intlData from "./messages";
+import {
+    Breadcrumbs,
+    Hidden,
+    Link,
+    List,
+    ListItem,
+    ListItemIcon,
+    ListItemText,
+    Menu,
+    Tooltip,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { injectIntl } from "react-intl";
+import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
+import ArrowRightIcon from "@material-ui/icons/ArrowRight";
+import FolderIcon from "@material-ui/icons/Folder";
+import { useRouter } from "next/router";
+import NavigationConstants from "../layout/NavigationConstants";
+import constants from "../../constants";
+import ids from "./ids";
+
+const useStyles = makeStyles((theme) => ({
+    selectedListItem: {
+        paddingLeft: 0,
+        color: theme.palette.primary.main,
+    },
+    list: {
+        outline: "none",
+        cursor: "pointer",
+        color: theme.palette.primary.main,
+        "&:hover": {
+            backgroundColor: theme.palette.primary.main,
+            color: theme.palette.primary.contrastText,
+        },
+    },
+    breadCrumbLink: {
+        textOverflow: "ellipsis",
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        display: "inline-block",
+        maxWidth: 150,
+    },
+    currentLocationLink: {
+        textOverflow: "ellipsis",
+        whiteSpace: "nowrap",
+        overflow: "hidden",
+        display: "inline-block",
+        maxWidth: 90,
+    },
+}));
+
+/**
+ * Splits a path into an array of path items.
+ * @example
+ * // returns [analyses,wordcount,logs]
+ * getPathItems("/analyses/wordcount/logs");
+ * @param {string }relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ * @returns {string[]} - path items as a string array  e.g: [analyses,wordcount,logs]
+ */
+function getPathItems(relativePath) {
+    let pathItems = [];
+    pathItems = relativePath.split("/").slice(1);
+    return pathItems;
+}
+
+/**
+ * Computes the path for routing.
+ *  * @example
+ * // returns "/data?path=/iplant/home/ipctest/analyses"
+ * getPathItems("/iplant/home/ipctest","/analyses/wordcount/logs", 0);
+ * @param {string }root - CyVerse Data Store root path e.g: /iplant/home/ipctest
+ * @param {string} relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
+ * @param {number} selectedPathItemIndex - index of the selected path item
+ * @returns {string} - path to be used by the nextjs router
+ */
+function pathToRoute(root, relativePath, selectedPathItemIndex) {
+    const relativePathItems = getPathItems(relativePath);
+    const reducer = (accumulator, currentVal, idx) => {
+        if (idx <= selectedPathItemIndex) {
+            return accumulator + "/" + currentVal;
+        }
+        return accumulator;
+    };
+    const routerPath = relativePathItems.reduce(reducer);
```

Comment:
This works, but I bet a combination of array `slice` and `join` might make it a little more concise.
```javascript
     const routerPath = [root, ...relativePath.slice(0, selectedPathItemIndex + 1)].join("/");
     return `/${NavigationConstants.DATA}?path=${routerPath}`;
```